server.go

```diff
@@ -197,6 +198,11 @@ func (sc *serverConn) handleStreams() {
 				sc.createStream(sc.c, strm)
 			}
 
+			if currentStrm != 0 && currentStrm != fr.Stream() {
+				sc.writeError(strm, NewGoAwayError(ProtocolError, "previous stream headers not ended"))
```

Comment:
I think this error will be triggered if you send a complete headers frame on one stream, and then another headers frame on another stream, and then data on the first stream. So:
```
-> Send Headers in stream = 1
-> Send Headers in stream = 3 <- will get rejected because we still processing the stream 1
-> Send Data in stream = 1
```

Am I right?

Reply:
If I am right, we need a flag to indicate that the headers ended and we do not expect any Data frame.

Reply:
You're right, I didn't read correctly, I thought the flag `endHeaderStream` (0x4) was handle by the `handleState`, but I was wrong. I will fix this ASAP.